package bakery;


import java.util.logging.Level;
import java.util.logging.Logger;
import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;


public class FoodOpenTest {
  FoodOpen f;


  public FoodOpenTest() {
  }

  @BeforeEach
  public void setUp() {
    try {
      f = new FoodOpen("KÃ¤se", 1, 14.60);
    }
    catch (BakeryException ex) {
      Logger.getLogger(FoodOpenTest.class.getName()).log(Level.SEVERE, null, ex);
    }
  }


  @Test
  public void testSetUnit() {
    try {
      f.setUnit(0);
      fail("Exception not thrown!");
    }
    catch (BakeryException ex) {
    }
    catch (Exception e) {
      fail("Invalid exception thrown!");
    }
  }


  @Test
  public void testSetPricePerUnit() {
    try {
      f.setPricePerUnit(0);
      fail("Exception not thrown!");
    }
    catch (BakeryException ex) {
    }
    catch (Exception e) {
      fail("Invalid exception thrown!");
    }
  }


  @Test
  public void testGetPrice() {
    assertEquals(14.6 * 1, f.calcPrice(), 0.001);
  }
}
